test/common/source/common.d

```diff
@@ -0,0 +1,57 @@
+module common;
+
+/// Name of the log file
+enum logFile = "test.log";
+
+/// has true if some test fails
+bool any_errors = false;
+
+/// prints (non error) message to standard output and log file
+void log(Args...)(Args args)
+	if (Args.length)
+{
+	import std.conv : text;
+	import std.stdio : File, stdout;
+
+	const str = text("[INFO] ", args);
+	version(Windows) stdout.writeln(str);
+	else stdout.writeln("\033[0;33m", str, "\033[0m");
+	stdout.flush;
+	File(logFile, "a").writeln(str);
+}
+
+/// prints error message to standard error stream and log file
+/// and set any_errors var to true value to indicate that some
+/// test fails
+void logError(Args...)(Args args)
+{
+	import std.conv : text;
+	import std.stdio : File, stderr;
+
+	const str = text("[ERROR] ", args);
+	version(Windows) stderr.writeln(str);
+	else stderr.writeln("\033[0;31m", str, "\033[0m");
+	stderr.flush;
+	File(logFile, "a").writeln(str);
+	any_errors = true;
+}
+
+auto die(string msg = "command failed", string supplemental = "", size_t line = __LINE__)
```

Comment:
```suggestion
void die(string msg = "command failed", string supplemental = null, size_t line = __LINE__)
```

Although I'm not sure why you don't pass `file` as well to get rid of `thisExePath` ?

Reply:
Well, actually, the `common.sh` design was because of bash's limitations. Nowadays `die` could just be:
```D
void die(Args...)(Args args)
{
    stderr.writeln(args);
    throw new Exception("Test failed");
}
```

Reply:
I would prefer to just port all bash scripts to D initially and then enhance them or redesign all test suit as needed. I have no time to make it at once, so this slow process step-by-step.

Reply:
Well bash script and D are quite different, so one can't just copy-paste the code and make it work with minor syntax changes like you would with C / C++. So it's not so much redesign here, it's just recognizing that the `die` function is meant to print an error message to `stderr` and exit. And doing this with varargs and Exception is much more natural to do in D than shoehorning the signature given to this function in bash.

But if you don't want to change it with this PR, I don't mind as long as the other comments are addressed.

Reply:
It is too defensive may be, but I try to make less intrusive changes as much as possible to avoid breaking something because I don't know code very well.  But if the reviewer says how  something can be improved I don't see the reason to resist it. That's what code review intended for.